docs/PowerAuth-Push-Server-1.4.0.md

```diff
@@ -0,0 +1,64 @@
+# Migration from 1.3.x to 1.4.x
+
+## New Message Inbox APIs
+
+You can review and consider implementing our [Message Inbox API](./Push-Server-API.md). Message inbox is a new feature
+of the push server that can be used to send targeted messages to the users to be read later.
+
+## Database Changes
+
+### Message Inbox Tables
+
+Message inbox requires a simple database structure below:
+
+#### PostgreSQL
+
+```sql
+CREATE TABLE push_inbox (
+    id VARCHAR(37) CONSTRAINT push_inbox_pk PRIMARY KEY,
+    user_id VARCHAR(255) NOT NULL,
+    subject TEXT NOT NULL,
+    body TEXT NOT NULL,
+    read BOOLEAN DEFAULT false,
+    timestamp_created TIMESTAMP NOT NULL,
+    timestamp_read TIMESTAMP
+);
+
+CREATE INDEX push_inbox_user ON push_inbox (user_id);
+CREATE INDEX push_inbox_user_read ON push_inbox (user_id, timestamp_created, read);
+```
+
+#### Oracle
+
+```sql
+CREATE TABLE PUSH_INBOX (
+  ID VARCHAR2(37 CHAR) PRIMARY KEY,
+  USER_ID VARCHAR2(255 CHAR) NOT NULL,
+  SUBJECT BLOB NOT NULL,
```

Comment:
I believe `CLOB` is the equivalent of `TEXT`. It is better for storing text and searching though it than binary data.